File: frontend-v2/src/services/businessDocument.ts
import api from './api'

// ============ Business Document (商务文档) ============

export interface BusinessDocumentSummary {
  id: string
  doc_type: string
  doc_name: string
  has_fillable_fields: boolean
  is_star_item: boolean
  score_point: string
  status: string
  source_file: string
}

export interface BusinessDocumentDetail extends BusinessDocumentSummary {
  project_id: string
  original_content: string
  editable_content: string
  rule_description: string
  return_file_list: string // JSON string
  source_file: string
}

export async function listBusinessDocuments(projectId: string): Promise<BusinessDocumentSummary[]> {
  const res = await api.get<BusinessDocumentSummary[]>(
    `/projects/${projectId}/business-documents`
  )
  return res.data
}

export async function getBusinessDocumentDetail(
  projectId: string,
  docId: string
): Promise<BusinessDocumentDetail> {
  const res = await api.get<BusinessDocumentDetail>(
    `/projects/${projectId}/business-documents/${docId}`
  )
  return res.data
}

export async function updateBusinessDocument(
  projectId: string,
  docId: string,
  payload: { editable_content?: string; status?: string }
): Promise<BusinessDocumentDetail> {
  const res = await api.patch<BusinessDocumentDetail>(
    `/projects/${projectId}/business-documents/${docId}`,
    payload
  )
  return res.data
}

export async function generateBusinessDocument(
  projectId: string,
  docId: string
): Promise<BusinessDocumentDetail> {
  const res = await api.post<BusinessDocumentDetail>(
    `/projects/${projectId}/business-documents/${docId}/generate`
  )
  return res.data
}

export interface DocumentExportResult {
  download_url: string
  filename: string
  format: string
}

export async function exportBusinessDocument(
  projectId: string,
  docId: string,
  fmt?: string
): Promise<DocumentExportResult> {
  const res = await api.post<DocumentExportResult>(
    `/projects/${projectId}/business-documents/${docId}/export`,
    null,
    { params: fmt ? { fmt } : undefined }
  )
  return res.data
}

export interface DocumentScoreResult {
  score: number
  max_score: number
  is_scored: boolean
  breakdown: Record<string, unknown>
  message?: string
}

export async function scoreBusinessDocument(
  projectId: string,
  docId: string
): Promise<DocumentScoreResult> {
  const res = await api.get<DocumentScoreResult>(
    `/projects/${projectId}/business-documents/${docId}/score`
  )
  return res.data
}

File: backend-v2/app/schemas/business_document.py
from pydantic import BaseModel, ConfigDict, Field


class BusinessDocumentSummary(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    doc_type: str
    doc_name: str
    has_fillable_fields: bool
    is_star_item: bool
    score_point: str
    status: str
    source_file: str


class BusinessDocumentDetail(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    project_id: str
    doc_type: str
    doc_name: str
    original_content: str
    editable_content: str
    has_fillable_fields: bool
    is_star_item: bool
    score_point: str
    rule_description: str
    status: str
    return_file_list: str  # JSON string
    source_file: str


class BusinessDocumentUpdateRequest(BaseModel):
    editable_content: str | None = Field(default=None)
    status: str | None = Field(default=None)


class DocumentExportResponse(BaseModel):
    download_url: str
    filename: str
    format: str


class ScoreBreakdown(BaseModel):
    completeness: float = Field(description="内容完整度 0-1")
    rule_match: float = Field(description="规则匹配度 0-1")
    semantic_quality: float = Field(description="语义质量 0-1 (LLM评估)")
    asset_coverage: float = Field(description="素材覆盖度 0-1")
    placeholder_count: int = Field(description="占位符数量")
    missing_keywords: list[str] = Field(default_factory=list, description="缺失关键词")
    llm_reasoning: str = Field(default="", description="LLM评分理由")


class DocumentScoreResponse(BaseModel):
    score: float = Field(description="当前得分")
    max_score: float = Field(description="满分")
    is_scored: bool = Field(description="是否已计分")
    breakdown: ScoreBreakdown = Field(description="评分明细")
    previous_score: float | None = Field(default=None, description="上次得分（用于对比）")
    score_delta: float | None = Field(default=None, description="得分变化")
    message: str | None = Field(default=None, description="提示信息")
